### PR TITLE
Feature/setup zod validation

### PR DESCRIPTION
**Pull request by aedoardo1990**

### CODE REVIEW

**Comment by aedoardo1990** [comment]
@lokytech5 I tested the messages of zod validation and they are displayed correctly in red. I have put on the first line of the page.tsx of the Register page the words "use client" because I was getting the error in the screenshot here attached

image

**Comment by lokytech5** [comment]
I understand. Most of the time, the annoying error with next.js is fixed by adding "use client" to the client component.

**Comment by aedoardo1990** [comment]
in your opinion was that the cause of failed deployment?

**Comment by lokytech5** [comment]
Yes, it is the cause of failed deployment. to by pass typescript-eslint, add [ // @ts-nocheck ] at the top of the register page.

File: frontend/app/register/page.tsx
Comment (lokytech5) [inline frontend/app/register/page.tsx:1]: add // @ts-nocheck at the top of this file to fix the deployment issue

Comment (lokytech5) [inline frontend/app/register/page.tsx:1]: add // @ts-nocheck at the top of this file to fix the deployment issue

Comment (aedoardo1990) [reply frontend/app/register/page.tsx:1]: @lokytech5 thanks! I have just updated the code as mentioned. I think the deployment is not working yet. If you want me to check more in detail to bug, could you pls grant me access to the deployment in vercel?

**Comment by lokytech5** [comment]
@aedoardo1990 can i have your email address so i can add you to my vercel team so you can go through the vercel deployment

**Comment by aedoardo1990** [comment]
@lokytech5 sure, my email address is edoardo.angeli1990@gmail.com

**Comment by lokytech5** [comment]
did you get the inivite

**Comment by lokytech5** [comment]
am going through the deployment error and it was referring to [Unexpected any. Specify a different type] meaning the type of the useForm data was not listed that is why typescript is throwing the error of unexpected any. there is a few adjustments that need to be done in the zodSchemas.ts file

**Comment by lokytech5** [comment]
previous code 
```bash
   
import * as z from 'zod';
export const registerSchema = z.object({
    name: z.string().min(2, 'Full name is required'),
    username: z.string().min(5, 'Username must be at least 5 characters'),
    email: z.string().email('Invalid email address'),
    password: z.string().min(8, 'Password must be at least 8 characters long'),
});
```
updated code with type for Formdata
```bash
import { z } from 'zod';
export const registerSchema = z.object({
    name: z.string().min(2, 'Full name is required'),
    username: z.string().min(5, 'Username must be at least 5 characters'),
    email: z.string().email('Invalid email address'),
    password: z.string().min(8, 'Password must be at least 8 characters long'),
});

export type FormData = z.infer<typeof registerSchema>;
```
the formData we then be imported to the register component and use below

```bash
import { registerSchema, FormData } from '../utils/zodSchemas';
const { register, handleSubmit, formState: { errors }, } = useForm<FormData>({ resolver: zodResolver(registerSchema),});
```

lastly remove this //@ts-nocheck.

**Comment by aedoardo1990** [comment]
morning @lokytech5! sorry for my late reply, I logged off yesterday. Yes I got the invite, thank you! I am checking the other annos you wrote now and I ll update the code

**Comment by aedoardo1990** [comment]
@lokytech5 thanks for the instructions to remove the bug, I added the code as mentioned and deploy failed again. I found this on stackoverflow (to add "/* eslint-disable @typescript-eslint/no-explicit-any */" - https://stackoverflow.com/questions/58467000/how-to-bypass-warning-unexpected-any-specify-a-different-type-typescript-eslin) and now deployment works

**Review by lokytech5 — APPROVED** [review]
Hello @aedoardo1990, am happy about the fix. thanks alot for this.

**Comment by lokytech5** [comment]
i we put this fix in mind ["/* eslint-disable @typescript-eslint/no-explicit-any */"] incase of future bug related to this.

**Comment by aedoardo1990** [comment]
happy to help! thanks a lot for all the tips! sure, I ll store it in a folder. Or is there maybe a shared platform where  you would like to save all the bugs fixes?

**Pull request merged**